browser/components/EditGroup.tsx

```diff
@@ -0,0 +1,302 @@
+/* Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import * as React from 'react';
+import SweetAlert from 'react-bootstrap-sweetalert';
+import { connect } from 'react-redux';
+import Select from 'react-select';
+
+import * as actions from '../actions/strategicActions';
+
+interface Props {
+  params: any;
+  fetchGroup: any;
+  fetchGroups: any;
+  fetchProjects: any;
+  fetchUsers: any;
+  updateGroup: any;
+  deleteGroup: any;
+  group: any;
+  projects: any;
+  users: any;
+  groupId: any;
+  updateAdminNav: any;
+}
+
+interface State {
+  name: string;
+  goal: string;
+  sponsor: string;
+  projects: any;
+  users: any;
+  alert: any;
+}
+
+class EditGroup extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      goal: '',
+      sponsor: '',
+      projects: [],
+      users: [],
+      alert: null,
+    };
+  }
+
+  componentDidMount() {
+    this.props.fetchGroup(this.props.groupId);
+    this.props.fetchProjects();
+    this.props.fetchUsers();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      const projects = [];
+      for (const proj of this.props.group.projects) {
+        projects.push(proj.project_id);
+      }
+      const users = [];
+      for (const user of this.props.group.users) {
+        users.push(user.amazon_alias);
+      }
+      this.setState({
+        name: this.props.group.group.group_name
+          ? this.props.group.group.group_name
+          : '',
+        goal: this.props.group.group.goal ? this.props.group.group.goal : '',
+        sponsor: this.props.group.group.sponsor
+          ? this.props.group.group.sponsor
+          : '',
+        projects: projects.join(','),
+        users: users.join(','),
+      });
+    }
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const fields = e.target.elements;
+    const projects = [];
+    const users = [];
+    if (fields.projectList.length > 1) {
```

Comment:
Add a few comments here describing what's happening. There are a few branches in here that make it dense to read through.